File: demo/java-demo/src/main/java/com/alibaba/testable/demo/service/DemoPrivateAccessService.java
package com.alibaba.testable.demo.service;

import org.springframework.stereotype.Service;

@Service
public class DemoPrivateAccessService {

    private int count;

    /**
     * private method
     */
    private String privateFunc(String s, int i) {
        return s + " - " + i;
    }

    /**
     * method with private field access
     */
    public String privateFieldAccessFunc() {
        count += 2;
        return String.valueOf(count);
    }

}

File: demo/kotlin-demo/src/main/kotlin/com/alibaba/testable/demo/service/DemoPrivateAccessService.kt
package com.alibaba.testable.demo.service

import org.springframework.stereotype.Service


@Service
class DemoPrivateAccessService {

    private var count = 0

    final val pi = 3.14

    /**
     * private method
     */
    private fun privateFunc(s: String, i: Int): String {
        return "$s - $i"
    }

    /**
     * method with private field access
     */
    fun privateFieldAccessFunc(): String {
        count += 2
        return count.toString()
    }

}
